dashboard/src/components/HandleLink/HandleLink.tsx

```diff
@@ -0,0 +1,34 @@
+import { LinkComponentProps } from 'node_modules/@tanstack/react-router/dist/esm/link';
+import React, { PropsWithChildren, useEffect, useRef, useState } from 'react';
+import { Link } from '@tanstack/react-router';
+
+const HandleLink: React.FC<PropsWithChildren<LinkComponentProps<'a'>>> = ({
+  children,
+  ...props
+}) => {
+  const [linkTarget, setLinkTarget] = useState<string | undefined>(undefined);
+  const linkRef = useRef<HTMLAnchorElement>(null);
```

Comment:
too many bloat for a simple link, it should be just use the Link component and that is it, no need for these hooks